shared/test/fake_redis_client.rb

```diff
@@ -29,8 +29,15 @@ def del(key)
   end
 
   def hdel(key, field)
-    value = get_hash_for_key(key).delete(field)
-    value ? 1 : 0
+    if field.is_a?(Array)
+      field.reduce(0) do |memo, next_field|
+        value = get_hash_for_key(key).delete(next_field)
+        memo + (value ? 1 : 0)
+      end
+    else
+      value = get_hash_for_key(key).delete(field)
```

Comment:
Perhaps you could avoid code duplication this to the array case by adding a line at the top of the method:
fields = [fields] unless fields.is_a?(Array)